R/check_pkg_installed.R

```diff
@@ -0,0 +1,180 @@
+#' Check Package Installation
+#'
+#' @description
+#' - `check_pkg_installed()`: checks whether a package is installed and
+#' returns an error or `FALSE` if not available. If a package search is provided,
+#' the function will check whether a minimum version of a package is required.
+#'
+#' - `get_pkg_dependencies()` returns a tibble with all
+#' dependencies of a specific package.
+#'
+#' - `get_min_version_required()` will return, if any, the minimum version
+#' of `pkg` required by `pkg_search`, `NULL` if no minimum version required.
+#'
+#' @param pkg (`string`)\cr
+#'   Package required
+#' @param call (`environment`)\cr
+#'   The execution environment of a currently running function. Default is `parent.frame()`.
+#'   This is used to message user about the original function call the resulted
+#'   in the prompt to install new packages.
+#' @param pkg_search (`string`)\cr
+#'   the package the function will search for a minimum required version from.
+#' @param return_lgl `(logical(1))`
+#'   logical indicating whether to return a `TRUE`/`FALSE`, rather
+#'   than error when package/package version not available. Default is `FALSE`,
+#'   which will return an error if `pkg` is not installed.
+#' @param remove_duplicates `(logical(1))`\cr
+#'   if several versions of a package are installed,
+#'   should only the first one be returned?
+#' @param lib.loc `(path)`
+#'   location of `R` library trees to search through, see
+#'   `utils::installed.packages()`.
+#' @details
+#' `get_all_packages_dependencies()` could be used to get the list of
```

Comment:
```suggestion
#' `get_all_pkg_dependencies(pkg_search = NULL)` could be used to get the list of
```

is that what you meant?

Reply:
ahh, thank you for reviewing so carefully!